internal/config/config.go

```diff
@@ -0,0 +1,58 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	"github.com/spf13/viper"
+	"github.com/subosito/gotenv"
+)
+
+func InitMainConfig() {
+	setDefaultValuesForMainConfig()
+	viper.SetConfigName(getMainConfigName()) // name of config file (without extension)
+	viper.SetConfigType("yaml")
+	viper.AddConfigPath(".")
+	_ = viper.ReadInConfig()
+
+	// Load env variables from .env
+	var err = gotenv.Load()
+	if err != nil {
+		fmt.Println("Gotenv load failed")
+	}
+
+	if os.Getenv("ENV") != "test" {
+		setupSentry()
+	}
+}
+
+func setDefaultValuesForMainConfig() {
+	viper.SetDefault("stats.interval", 30)
+	viper.SetDefault("log.level", "error")
+}
+
+// depending on ENV variable creates name for config file
+func getMainConfigName() string {
+	configFileName := "config"
+	if env := os.Getenv("ENV"); env != "" {
+		configFileName = configFileName + "-" + env
+	}
+
+	return configFileName
+}
+
+func setupSentry() {
+	dsn := os.Getenv("SENTRY_DSN")
```

Comment:
At some point, we should refactor this to use only config variables, to have one source for all needed variables.

Reply:
https://github.com/NodeFactoryIo/vedran-daemon/issues/8